### PR TITLE
Update web\css area to use latest `rgb()` and `hsl()` syntax

### PR DESCRIPTION
**Pull request by FormularSumo**
This PR replaces all `rgba()`/`hsla()` occurrences with `rgb()`/`hsl()` and updates all `rgb()`/`hsl()` occurrences to use the latest syntax, within files\en-us\web\css

See #30844 for more details on why these changes are being made.

There were a few places where rgba/hsla can still be found, for legacy reasons or other usage. These are:
- color
- color_value
- css_colors
- css_color\applying_color
- css_values_and_units
- hex-color

Additionally, there were a couple of other minor related changes I made:
- In css_images\using_css_gradients lines 610-613 had an extra indent, which I've removed (edit: I was wrong about this, undone)
- In css_colors\applying_color both the old and new syntax ways of describing the alpha value (number vs percentage) were used, which I've changed to be just the new one (edit: I've partially reverted this, now describes percentage and then number)

### CODE REVIEW

**Comment by FormularSumo** [comment]
In this commit I've been replacing numbers with percentages as the related issue said that is new syntax. The formal syntax on the rgb and hsl pages seems to say that the alpha value being a number isn't modern vs legacy, it's just an option. Both pages use the number variant more often. If they're wrong and percetange is preferred, then they need updating I think. If they're correct and it doesn't matter, than I need to undo my change to css_colors\applying_color and in future won't change percentages to numbers unless it happens to be more convenient to (eg sometimes when replacing rgba/hsla values)

File: files/en-us/web/css/css_images/using_css_gradients/index.md [hunk 1]
```diff
@@ -750,24 +750,24 @@ div {
...
       -190deg,
-      rgba(255, 0, 0, 0.5) 30px,
-      rgba(255, 153, 0, 0.5) 60px,
-      rgba(255, 255, 0, 0.5) 90px,
-      rgba(0, 255, 0, 0.5) 120px,
-      rgba(0, 0, 255, 0.5) 150px,
-      rgba(75, 0, 130, 0.5) 180px,
-      rgba(238, 130, 238, 0.5) 210px,
-      rgba(255, 0, 0, 0.5) 230px
+      rgb(255 0 0 / 50%) 40px,
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_images/using_css_gradients/index.md:763]: The color values look correct, but you've changed the length unit value in each case. Was that intentional?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
Hi there @FormularSumo, and thanks so much for this excellent contribution to MDN! I've found a few minor syntax bits for you to fix up, and I've commented on the changes to the "Applying color" article that didn't think quite work in their present state. Apart from that, it is all looking great; I like the idea of normalizing on % alpha values as a general rule.

**Comment by chrisdavidmills** [comment]
@FormularSumo OK, reviewing finished! I noticed that you've got a linter failure in the checks list too. This should be fixed by accepting the changes the mdn linter has suggested above. If not, then a sure-fire way to fix this error is to run prettier on all the files you've changed in the repo, and pushing the result.

So for example, by running `npx prettier -w **/*.md` in the repo root, or however you want to handle it.

File: files/en-us/web/css/css_images/using_css_gradients/index.md (hunk 1 above)
Comment (FormularSumo) [reply files/en-us/web/css/css_images/using_css_gradients/index.md]: Nope sorry, I didn't realise they were different. I've changed them back to what they were originally now.

File: files/en-us/web/css/border-color/index.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ border-color: red;
 border-color: red #f015ca;
```
[the remark this answers is not in the document]
Comment (FormularSumo) [reply files/en-us/web/css/border-color/index.md:34]: It is an example of that I think, I don't believe it has been removed though? I think the #f015ca; is only showing up as red it's a hex colour value and github decides to highlight them (slightly confusingly).

[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/border-color/index.md:34]: Ack, sorry about that — yes, that's me misreading the GitHub diff! You can ignore this one.

File: files/en-us/web/css/css_colors/applying_color/index.md [hunk 3]
```diff
@@ -118,9 +118,9 @@ Legal values for each of these parameters are:
 - `red`, `green`, and `blue`
   - : Each must be an {{cssxref("&lt;integer&gt;")}} value between 0 and 255 (inclusive), or a {{cssxref("&lt;percentage&gt;")}} from 0% to 100%.
 - `alpha`
-  - : The alpha channel is a number between 0.0 (fully transparent) and 1.0 (fully opaque). You can also specify a percentage where 0% is the same as 0.0 and 100% is the same as 1.0.
+  - : The alpha channel is percentage where 0% (fully transparent) and 100% (fully opaque).
```
[the remark this answers is not in the document]
Comment (FormularSumo) [reply files/en-us/web/css/css_colors/applying_color/index.md]: https://github.com/mdn/content/pull/31453/commits/fae86b38a7598561b800c61f897898f984f9b036 I've mostly undone this change, but put percentage first and number second. Similarly, I've updated rgb and hsl to use both, percentage on the new syntax, and number on the old, so examples of both are used. The only potential issue I see is people reading them might assume you can only use percentage with the new syntax and number with old; but I think it's still better than only giving an example of just only one of them.

**Comment by FormularSumo** [comment]
I believe I've made all the requested changes, including linter ones. I'm not 100% sure about the ones in applying color, rgb, and hsl but otherwise I'm pretty happy with everything. Thanks for all the feedback!

File: files/en-us/web/css/css_colors/applying_color/index.md [hunk 4]
```diff
@@ -118,9 +118,9 @@ Legal values for each of these parameters are:
 - `red`, `green`, and `blue`
   - : Each must be an {{cssxref("&lt;integer&gt;")}} value between 0 and 255 (inclusive), or a {{cssxref("&lt;percentage&gt;")}} from 0% to 100%.
 - `alpha`
-  - : The alpha channel is a number between 0.0 (fully transparent) and 1.0 (fully opaque). You can also specify a percentage where 0% is the same as 0.0 and 100% is the same as 1.0.
+  - : The alpha channel is percentage between 0% (fully transparent) and 100% (fully opaque), or a number between 0.0 (same as 0%) and 1.0 (same as 100%).
```
Comment (chrisdavidmills) [inline files/en-us/web/css/css_colors/applying_color/index.md:121]:
```suggestion
  - : The alpha channel is specified as a percentage between 0% (fully transparent) and 100% (fully opaque), or a number between 0.0 (equivalent to 0%) and 1.0 (equivalent to 100%).
```

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@FormularSumo one more suggestion; we are very close now!

**Review by chrisdavidmills — APPROVED** [review]
@FormularSumo. Ding ding, I think we're done! Approving. Thanks again for the wonderful contribution.

**Pull request merged**